File: src/main/scala/org/bakkes/game/PathsModule.java
package org.bakkes.game;

import java.nio.file.FileSystems;
import java.nio.file.Path;

import org.bakkes.game.model.map.IAreaNameAcces;

import com.google.inject.Provides;
import com.google.inject.name.Named;
import com.google.inject.name.Names;

public class PathsModule extends AModule {
	@Override
	public void configure(){
		final Path base = FileSystems.getDefault().getPath("bin");
		bind(Path.class).toInstance(base);
        final Path scripts = bind(base, "scripts");
        bind(scripts, "pokemon","scriptPokemon");
        bind(scripts, "moves");
        bind(scripts, "items", "scriptItems");
        final Path overworld = bind(scripts, "overworld");
        bind(overworld, "areas");
        bind(overworld, "animations");
        final Path sprites = bind(base, "sprites");
        bind(sprites, "pokemon", "spritePokemon");
        bind(sprites, "items", "spriteItems");
        bind(base, "map");
	}
	private Path bind(final Path base, final String dirName){
		return bind(base,dirName,dirName);
	}
	private Path bind(final Path base, final String dirName, final String annoName){
        final Path result = base.resolve(dirName);
        bind(Path.class).annotatedWith(Names.named(annoName)).toInstance(result);
        return result;
	}
	@Provides
	@Named("current area")
	public Path getScriptFolder(final IAreaNameAcces named, @Named("areas") final Path path){
		return path.resolve(named.getAreaName());
	}
}

File: src/main/scala/org/bakkes/game/PathsModule.scala
package org.bakkes.game

import java.nio.file.{FileSystems, Path}

import com.google.inject.Provides
import com.google.inject.name.{Named, Names}
import net.codingwell.scalaguice.ScalaModule
import org.bakkes.game.model.map.IAreaNameAcces

class PathsModule extends AModule with ScalaModule{

	override def configure {
		//val base = FileSystems.getDefault.getPath("build").resolve("resources").resolve("main") // production?
		//val base = FileSystems.getDefault.getPath("bin") //eclipse
		val base = FileSystems.getDefault.getPath("out").resolve("production").resolve("Slick2DRPG") // intelij
		bind[Path].toInstance(base)
		val scripts = bind(base, "scripts")
		bind(scripts, "pokemon", "scriptPokemon")
		bind(scripts, "moves")
		bind(scripts, "items", "scriptItems")
		val overworld = bind(scripts, "overworld")
		bind(overworld, "areas")
		bind(overworld, "animations")
		val sprites = bind(base, "sprites")
		bind(sprites, "pokemon", "spritePokemon")
		bind(sprites, "items", "spriteItems")
		bind(base, "map")
	}

	private def bind(base: Path, dirName: String): Path = bind(base, dirName, dirName)

	private def bind(base: Path, dirName: String, annoName: String): Path = {
		val result: Path = base.resolve(dirName)
		bind[Path].annotatedWith(Names.named(annoName)).toInstance(result)
		return result
	}

	@Provides
	@Named("current area") def getScriptFolder(named: IAreaNameAcces, @Named("areas") path: Path): Path = {
		return path.resolve(named.getAreaName)
	}
}